### PR TITLE
#85 employee page

### PR DESCRIPTION
**Pull request by fatmaalzharaa**

### CODE REVIEW

File: retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.html [hunk 1]
```diff
@@ -10,110 +10,73 @@ <h3 class="panel-title fw-bold">
...
+            </button>
+          </mat-cell>
+        </ng-container>
+        <mat-header-row *matHeaderRowDef="displayedColumns;sticky:true;"></mat-header-row>
+        <mat-row *matRowDef="let row; columns: displayedColumns;" ></mat-row>
+      </mat-table>
+    
+        <mat-paginator [length]="paginationEmployee?.itemCount"
+          [pageSize]="paginationEmployee?.size"
+          [pageSizeOptions]="[5, 10,20]"
```
[the remark this answers is not in the document]
Comment (fatmaalzharaa) [reply retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.html]: done

File: retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.html [hunk 2]
```diff
@@ -10,110 +10,73 @@ <h3 class="panel-title fw-bold">
...
           (page)="pageEvent=$event;onPaginationChange($event)"
           showFirstLastButtons
-          pageIndex=""
           aria-label="Select page"
-          class="fw-bold fs-6 pagination ">
+          class="fw-bold fs-6 text-info pagination pe-4 me-2 py-3 ">
           </mat-paginator>
 
       </div>
```
[the remark this answers is not in the document]
Comment (fatmaalzharaa) [reply retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.html:82]: done

**Review by AsmaMohame — APPROVED** [review]

**Review by AsmaMohame — APPROVED** [review]

File: retail-managment-system/rms-ui/package.json [hunk 3]
```diff
@@ -24,6 +24,7 @@
     "@ngx-translate/http-loader": "^7.0.0",
     "bootstrap": "^5.2.2",
     "material-design-icons": "^3.0.1",
+    "ngx-pagination": "^6.0.3",
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/package.json:27]: are you use this package?

File: retail-managment-system/rms-ui/src/app/core/models/response.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { Pagination } from "./pagination";
+export interface response<T>{
+  data: [T];
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/models/response.ts:3]: [T] or T[]

File: retail-managment-system/rms-ui/src/app/core/models/response.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { Pagination } from "./pagination";
+export interface response<T>{
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/models/response.ts:2]: name of interface must be pascal case

File: retail-managment-system/rms-ui/src/app/core/services/resource.service.ts [hunk 6]
```diff
@@ -90,4 +90,18 @@ export abstract class ResourceService<T> {
         })
       );
   }
+
+  findAll(page:number,size:number): Observable<response<T>>{
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/services/resource.service.ts:94]: why add new function?
why didn't use getList()?

File: retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.ts [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    'primaryPhoneNo',
+    'secondaryPhoneNo',
+    'address',
+    'job',
+    'actions',
+  ];
+  constructor(
+    private employeesRepository: EmployeeRepository,
+    private build: FormBuilder,
+    private _snackBar: MatSnackBar,
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.ts:39]: use camel case

File: retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.ts [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    this.pagination()
+  }
+  getAllEmployee() {
+    this.employeesRepository.getList().subscribe((result) => {
+      this.dataSource = result.data;
+    });
+  }
+
+  pagination() {
+    this.employeesRepository.getList().subscribe((result) => {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.ts:56]: why add new function to get pagination?
use same function getAllEmployee()

File: retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.ts [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+      this.paginationEmployee = result.pagination;
+    });
+  }
+  onPaginationChange(event: PageEvent) {
+    let page = event.pageIndex;
+    let size = event.pageSize;
+    this.employeesRepository.findAll(page, size).pipe(
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.ts:63]: you already have function that get employee why make another?
call function getAllEmployee()

File: retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.html [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+        <mat-header-row *matHeaderRowDef="displayedColumns;sticky:true;"></mat-header-row>
+        <mat-row *matRowDef="let row; columns: displayedColumns;" ></mat-row>
+      </mat-table>
+
+        <mat-paginator [length]="paginationEmployee?.itemCount"
+          [pageSize]="paginationEmployee?.size"
+          [pageSizeOptions]="[5, 10,25,100]"
+          (page)="pageEvent=$event;onPaginationChange($event)"
+          showFirstLastButtons
+          [dir]="'direction-pagination'|translate"
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/employee/employee/employee.component.html:78]: direction is same in ar and en so you can delete this

File: retail-managment-system/rms-ui/src/assets/i18n/ar.json [hunk 11]
```diff
@@ -85,13 +87,15 @@
   },
   "customrs": {
     "customer": "العملاء",
+    "employee": "موظف",
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/assets/i18n/ar.json:90]: add section for employee don't use costumers 

as structure we make in translation

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/package.json (hunk 3 above)
Comment (fatmaalzharaa) [reply retail-managment-system/rms-ui/package.json:27]: yes

**Comment by ZinabHashem** [comment]
- add card for employee in home page.
- use total row com from back in request to show this number in pagination after add or delete.

File: retail-managment-system/rms-ui/package.json (hunk 3 above)
Comment (ZinabHashem) [reply retail-managment-system/rms-ui/package.json:27]: where?

**Comment by ZinabHashem** [comment]
@AsmaMohame review again plz

File: retail-managment-system/rms-ui/src/assets/i18n/ar.json [hunk 12]
```diff
@@ -109,6 +112,30 @@
     "delete-successfuly": "تم مسح العميل بنجاح",
     "close": "اغلاق"
   },
+  "Employee": {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/assets/i18n/ar.json:115]: why start with capital character? 
use same case we use here

File: retail-managment-system/rms-ui/src/app/shared/material-module/material-module.module.ts [hunk 13]
```diff
@@ -41,6 +41,7 @@ import { MatTooltipModule } from '@angular/material/tooltip';
 import { MatPaginatorModule } from '@angular/material/paginator';
 import { MatSortModule } from '@angular/material/sort';
 import { MatTableModule } from '@angular/material/table';
+import { NgxPaginationModule } from 'ngx-pagination';
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/shared/material-module/material-module.module.ts:44]: NgxPaginationModule  is exist in project?

File: retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.html [hunk 14]
```diff
@@ -6,11 +6,6 @@
       "sidebar.home" | translate
     }}</a>
   </p>
-  <p>
-    <a mat-button routerLink="order" routerLinkActive="active" class="fs-5">{{
-      "sidebar.order" | translate
-    }}</a>
-  </p>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/layout/side-bar/side-bar.component.html:13]: why delete this ?

**Pull request closed without merging**